Expensify/App
# Attachment cycling with onyx

Implemented functionality to achieve the following statement:
"The attachments present in the chat should cycle when the user is viewing the modal and presses the left/right arrows on their keyboard"

### Details
Refactored code and moved Attachment modal so only one is created for ReportActionsView and one for the ReportActionCompose view. AttachmentModal now consumed visibleModal object from Onyx to know what to show and wether to be visible.

Also wrote getNextAttachment function which retrieves the next attachment using the current action and a boolean value indicating whether to move to the right or left of the current action.

### Fixed Issues
Implements #1038 

### Tests

1. Upload multiple attachments (Images and PDFs)
2. Once they have been uploaded, use the left and right arrow keys to move between all of them.
3. Go to the last attachment and confirm that moving to the right of the last attachment takes you to the first attachment in a cyclical manner.

1. Test that uploading attachments still work (Images, PDFs .etc)

### Tested On

### Screenshots

#### Web
<img width="2048" alt="Screenshot 2020-12-31 at 19 33 10" src="https://user-images.githubusercontent.com/13122879/103423562-25bda180-4b9f-11eb-9b68-22d2b26b6795.png">
<img width="2048" alt="Screenshot 2020-12-31 at 19 33 27" src="https://user-images.githubusercontent.com/13122879/103423566-2a825580-4b9f-11eb-9328-d3f53a567e82.png">

#### Mobile Web
no changes

#### Desktop
no changes

#### iOS
no changes

#### Android
no changes
